airflow/www/utils.py

```diff
@@ -302,6 +311,48 @@ def dag_run_link(attr):
         '<a href="{url}">{run_id}</a>').format(url=url, run_id=run_id)  # noqa
 
 
+def dag_query_for_key(sorting_key):
+    """Maps sorting key param in the URL params to DB queries on appropriate attributes."""
+    dag_query_key_map = {
+        'dag_id': DagModel.dag_id,
+        'owner': DagModel.owners,
+        'schedule': DagModel.next_dagrun,
+        # <- add any extra (URL param)->(DagModel attr) mappings here
+    }
```

Comment:
It might be easier/less code to have a single `sort_by` and have this as

```suggestion
    dag_query_key_map = {
        'dag_id': DagModel.dag_id,
        '-dag_id': DagModel.dag_id.desc(),
        'owner': DagModel.owners,
        '-owner': DagModel.owners.desc(),
        'schedule': DagModel.next_dagrun,
        '-schedule': DagModel.next_dagrun.desc(),
        # <- add any extra (URL param)->(DagModel attr) mappings here
    }
```

That way we don't need to have extra sorting_order param. I think this makes the code and the URI simpler.

Additionally this order probably doesn't belong in this file -- but instead in the views.py file, probably in the `home` function.

Reply:
(If you think the current way is clearer, tell me. But I do think the location is wrong for this, as it is specific to one particular view.)

Reply:
I see your point, but I feel like the suggested way is a bigger PITA from a long-term maintenance perspective - you have to add each new sortable in duplicate. It also seems less intuitive when looking at the URL - the current setup basically just replicates the underlying SQL. All in all, I was trying to make the two bits as independently testable as possible.

I may be getting ahead of myself, but it should also make things easier if someone wanted to sort by multiple columns - e.g. ASC by ID and DESC by owner would be `orderBy=asc;desc` and then you could just `zip()` it with `sortBy`'s items.

Re: the module location, point taken. I will move things around as appropriate. I was kind of hoping someone would slap me and point me to the right place if this one wasn't it, so - appreciated!